core/slice.go

```diff
@@ -697,7 +700,11 @@ func (sl *Slice) ConstructLocalBlock(header *types.Header) *types.Block {
 			block = types.NewBlockWithHeader(header).WithBody(txs, uncles, etxs, subBlockHashes)
 		}
 	}
-	return block
+	if err := sl.validator.ValidateBody(block); err != nil {
+		return block, err
```

Comment:
we should probably not return the block here. Just nil, err.

Reply:
The reason to return the block, is because we may want to trap the error and use the block anyways.

e.g. if the error is 'ErrBadSubManifest', we may just want to fill the block's sub manifest from our subordinate chain and keep going.

Reply:
Okay. it doesn't look like we do anything with the block at the moment. we just return the error.

Reply:
I have another branch which may have to do that... still in flight.

Regardless, does it matter? If you receive an `err` value, you should always handle the error. The simplest form of handling that we do in 99% of cases is to abort and return the error; i.e. ignoring the value.

Reply:
okay. thats good. only reason i pointed this out is because if people use this function and in the case of error if they use the block for something nothing will break. But if there is a change incoming where we use the value I am fine with it.